lib/typed_buffers.dart

```diff
@@ -213,8 +213,6 @@ abstract class _TypedDataBuffer<E> extends ListBase<E> {
       if (start > values.length || end > values.length) {
         throw StateError('Too few elements');
       }
-    } else {
-      assert(end != null);
```

Comment:
Do we want to keep some of these in for mixed mode tests?

Reply:
We effectively get the assert because we're doing math below (`end - start`)

I think this is fine to ditch!